### PR TITLE
valhalla jdk-23+1 bringup

### PR DESCRIPTION
**Pull request by JasonFengJ9**
valhalla jdk-23+1 bringup

Removed `jdk.internal.value.PrimitiveClass` references, and updated `Access` for APIs;
Removed `Modifier.VALUE` references;
Renamed `Unsafe` methods `isFlattenedArray()` to `isFlatArray()`, `isFlattened()` to `isFlatField()`, and updated the references;
Added stub methods `JVM_IsImplicitlyConstructibleClass()`, `JVM_IsNullRestrictedArray()` and `JVM_NewNullRestrictedArray()`;
Renamed `MN_FLATTENED` to `MN_FLAT_FIELD` according to `java.lang.invoke.MethodHandleNatives`, also added `MN_NULL_RESTRICTED`, and updated `MN_REFERENCE_KIND_SHIFT` and `MN_REFERENCE_KIND_MASK`;
Added `fieldalias` for `java.lang.reflect.Field.flags`.

Required by
* https://github.com/ibmruntimes/openj9-openjdk-jdk.valuetypes/pull/10
* https://github.com/ibmruntimes/openj9-openjdk-jdk.valuetypes/pull/11

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Comment by hangshao0** [comment]
@theresa-m, this is the change required in OpenJ9 to let us merge in the latest Valhalla change in the extension repo. I've added you as reviewer. 

Once this change is merged, there might be some updates needed for Valhalla tests which are based on the obsolete spec. You mighty have already identified or fixed some of them when working with lw5 before.

**Review by theresa-m — CHANGES_REQUESTED** [review]

File: jcl/src/java.base/share/classes/java/lang/Class.java [hunk 1]
```diff
@@ -2853,16 +2853,12 @@ public String toGenericString() {
 
 /*[IF INLINE-TYPES]*/
 	String valueType;
-	if (Modifier.VALUE == (Modifier.VALUE & modifiers)) {
+	if (isValue()) {
 		/**
 		 * Modifier.PRIMITIVE is not added by OpenJDK yet. It shares the same bit as Modifier.STRICT.
```
[the remark this answers is not in the document]
Comment (JasonFengJ9) [reply jcl/src/java.base/share/classes/java/lang/Class.java]: Removed as well.

File: jcl/src/java.base/share/classes/java/lang/Class.java [hunk 2]
```diff
@@ -2895,7 +2891,7 @@ public String toGenericString() {
 	 * Modifier.PRIMITIVE is not added by OpenJDK yet. It shares the same bit as Modifier.STRICT.
```
[the remark this answers is not in the document]
Comment (JasonFengJ9) [reply jcl/src/java.base/share/classes/java/lang/Class.java]: Also removed.

File: jcl/src/java.base/share/classes/java/lang/Object.java [hunk 3]
```diff
@@ -275,11 +275,11 @@ public final void wait(long time) throws InterruptedException {
  */
 public final void wait(long time, int frac) throws InterruptedException {
 /*[IF JAVA_SPEC_VERSION >= 19]*/
-/*[IF JAVA_SPEC_VERSION >= 23]*/
+/*[IF (JAVA_SPEC_VERSION >= 23) & !INLINE-TYPES]*/
 	boolean blockerRC = Blocker.begin();
```
[the remark this answers is not in the document]
Comment (JasonFengJ9) [reply jcl/src/java.base/share/classes/java/lang/Object.java:279]: This change was to accommodate [a recent openjdk update](https://github.com/ibmruntimes/openj9-openjdk-jdk/commit/e592d365bdcf133bc740531eba65f5bc01cedd0d) which hasn't been integrated into   [valhalla](https://github.com/openjdk/valhalla) and [openj9-openjdk-jdk.valuetypes](https://github.com/ibmruntimes/openj9-openjdk-jdk.valuetypes) yet.

File: jcl/src/java.base/share/classes/java/lang/Class.java [hunk 4]
```diff
@@ -2892,10 +2879,8 @@ public String toGenericString() {
 	 * IDENTITY shares the same bit as SYNCHRONIZED. VALUE shares the same bit as VOLATILE.
```
Comment (theresa-m) [inline jcl/src/java.base/share/classes/java/lang/Class.java:2879]: Last comment nitpick is the mention on VALUE here.

**Review by theresa-m — APPROVED** [review]

File: jcl/src/java.base/share/classes/java/lang/Class.java (hunk 4 above)
Comment (JasonFengJ9) [reply jcl/src/java.base/share/classes/java/lang/Class.java]: Removed the `VALUE` reference.

**Comment by hangshao0** [comment]
Jenkins compile zlinuxval jdknext depends https://github.com/ibmruntimes/openj9-openjdk-jdk.valuetypes/pull/11

**Comment by hangshao0** [comment]
Jenkins compile winval jdknext depends https://github.com/ibmruntimes/openj9-openjdk-jdk.valuetypes/pull/11

**Comment by JasonFengJ9** [comment]
Resolved the merging conflict.

**Comment by hangshao0** [comment]
Jenkins compile zlinux jdknext

**Comment by hangshao0** [comment]
Launched an internal Valhalla build to test the changes. There is a failure. @JasonFengJ9 is looking at that.

**Comment by JasonFengJ9** [comment]
```
18:04:59  /home/jenkins/workspace/Build_JDKnext_x86-64_linux_valhalla_Personal/test/hotspot/jtreg/serviceability/jvmti/HiddenClass/libHiddenClassSigTest.cpp:208:8: error: 'found' was not declared in this scope
18:04:59    208 |   if (!found) {
```
Failed at test code compilation.

**Comment by JasonFengJ9** [comment]
Launched [an internal personal build](https://hyc-runtimes-jenkins.swg-devops.com/view/OpenJ9%20-%20Personal/job/Pipeline-Build-Test-Personal/21932/)

**Review by hangshao0 — APPROVED** [review]
LGTM.

**Comment by hangshao0** [comment]
FYI @hzongaro @a7ehuo  
With this PR and https://github.com/ibmruntimes/openj9-openjdk-jdk.valuetypes/pull/11 merged, we brought in the latest Valhalla changes from OpenJDK.

File: runtime/oti/j9nonbuilder.h [hunk 5]
```diff
@@ -2294,11 +2294,14 @@ typedef struct J9ROMMethodHandleRef {
 #define MN_TRUSTED_FINAL	0x00200000
 #define MN_HIDDEN_MEMBER	0x00400000
 #if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
-#define MN_FLATTENED		0x00800000
-#endif /* defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES) */
-
+#define MN_FLAT_FIELD		0x00800000
+#define MN_NULL_RESTRICTED	0x01000000
+#define MN_REFERENCE_KIND_SHIFT		26
+#define MN_REFERENCE_KIND_MASK		0x3C000000 >> MN_REFERENCE_KIND_SHIFT	/* (flag >> MN_REFERENCE_KIND_SHIFT) & MN_REFERENCE_KIND_MASK */
```
Comment (keithc-ca) [inline runtime/oti/j9nonbuilder.h:2300]: Please add parentheses here:
```c
#define MN_REFERENCE_KIND_MASK		(0x3C000000 >> MN_REFERENCE_KIND_SHIFT))
```
On the other hand, the result is the same (`0xF`) when `!defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)`: why make the definition conditonal?

Comment (JasonFengJ9) [reply runtime/oti/j9nonbuilder.h:2300]: > On the other hand, the result is the same (0xF) when !defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES): why make the definition conditional?

As per https://github.com/eclipse-openj9/openj9/blob/2c34fcf0fb16edf7ff0c5af17db501e6107cb39f/runtime/oti/j9nonbuilder.h#L2285-L2286
This maps https://github.com/ibmruntimes/openj9-openjdk-jdk.valuetypes/blob/12c6a4c77383821a4499e5472d7c4241542d546f/src/java.base/share/classes/java/lang/invoke/MethodHandleNatives.java#L126-L129
```
            MN_FLAT_FIELD            = 0x00800000, // flat field
            MN_NULL_RESTRICTED       = 0x01000000, // null-restricted field
            MN_REFERENCE_KIND_SHIFT  = 26, // refKind
            MN_REFERENCE_KIND_MASK   = 0x3C000000 >> MN_REFERENCE_KIND_SHIFT;
```

Added parentheses at 
* https://github.com/eclipse-openj9/openj9/pull/19446

**Pull request merged**